### ISSUE TITLE
Transition matrix stabilization results in un-normalized transition matrices

### ISSUE THREAD

**Issue by ckemere** [hmmlearn/hmmlearn#65]
In base.py, the transition matrix is calculated from the sufficient statistics and then normalized. However, in the follow line, the transition matrix update is done element-by-element, ignoring cells which are less than eps. The result is an un-normalized transition matrix if fit() stops after this update, which breaks, e.g., score().

I just eliminated the eps - check, which means that some cells of the transition matrix will effectively go to zero. I'm not sure what the "right" thing to do is.

````
        if 't' in params:
            transmat_ = self.transmat_prior - 1.0 + stats['trans']
            normalize(transmat_, axis=1)
            self.transmat_ = np.where(self.transmat_ <= np.finfo(float).eps,
                                      self.transmat_, transmat_)
````

**Comment by superbobry** [comment]
Hello, thank you for reporting. This does seem like a problem.

**Comment by superbobry** [comment]
I gave his some more thought and now I'm not sure if the `<= eps` comparison is doing any trouble. It was the added to allow for "impossible" (i.e. having zero probability) states or transitions. 

For the normal case, when all transition probabilities are none-zero, it seems highly unlikely that during training some transition the probability will underflow to zero. What do you think?

**Comment by ckemere** [comment]
It should presumably sum to one regardless?

**Comment by superbobry** [comment]
It should, and I believe it will, yes.

**Comment by ckemere** [comment]
In the current approach, doing anything after normalize will potentially cause it not to be one. So if you want to maintain a floor value, a bit more code is needed.

**Comment by superbobry** [comment]
Yes, sorry, of course you're right. I'm working on a proper fix.

**Comment by superbobry** [comment]
This has been fixed in 144dc5318b83b2385cb14239ef7b30b8e833f902 and tested in 1edac767592f7419d0e7f86f5ea1686a0d4fb41d.
